public/scripts/report.js

```diff
@@ -1,24 +1,80 @@
 (function(){
 
-  // GET LOCATION
-  var locationButton = document.getElementById('getLocationButton');
-
-  function getLocation() {
-    if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition(showPosition);
-    } else {
-      // non-supprt error handling
-      console.log('Geolocation is not supported by this browser.');
+  // GET USER'S LOCATION
+  function getLocation(){
```

Comment:
You've set this to execute immediately at the bottom of the file - instead we just want to apply the event listener, not run the whole getLocation function. Maybe separate it out like you did with addSubmitListener and extractFormData. Probably good to just have one function which adds both event listeners, and call that immediately at the start.